daisy_workflows/build-publish/rhui/health_check.py

```diff
@@ -0,0 +1,136 @@
+#  Copyright 2022 Google Inc. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http:#www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Health check for RHUI infrastructure nodes."""
+
+import argparse
+import logging
+import logging.handlers
+import os
+import sys
+import tempfile
+import typing
+
+
+class HealthChecks:
+  nfs_mount: str
+
+  def __init__(
+      self,
+      nfs_mount: str,
+  ) -> None:
+    self.nfs_mount = nfs_mount
+
+  def mount_uses_nfs(self, logger: logging.Logger):
+    logger.info("checking that %s is mounted as NFS", self.nfs_mount)
+    with open("/proc/mounts") as mounts:
+      for line in mounts:
+        parts = line.split()
+        mnt_point, mnt_type = parts[1], parts[2]
+        if mnt_point == self.nfs_mount:
+          if mnt_type == "nfs":
+            return
+          else:
+            raise EnvironmentError("Not mounted as nfs. mount: %s" % line)
+    raise EnvironmentError("%s not in /proc/mounts" % self.nfs_mount)
+
+  def mount_is_writable(self, logger: logging.Logger):
+    logger.info("checking that %s is writable", self.nfs_mount)
+    with tempfile.TemporaryFile(dir=self.nfs_mount) as f:
+      f.write(b"content")
+
+  def mount_is_readable(self, logger: logging.Logger):
+    logger.info("checking that %s is readable", self.nfs_mount)
+    os.listdir(self.nfs_mount)
+
+
+def main(node_type: str, result_file: typing.TextIO, nfs_mount: str):
+  """Executes health check for node_type, and writes the overall
+  result to result_file ('OK' if all checks pass, otherwise 'ERROR').
+  """
+  health_checks = HealthChecks(nfs_mount=nfs_mount)
+  checks = [
+    health_checks.mount_is_readable,
+    health_checks.mount_uses_nfs,
+  ]
+  if node_type == "rhua":
+    checks += [
+      health_checks.mount_is_writable,
+    ]
+  success = True
+  for func in checks:
+    logger = logging.getLogger(func.__name__)
+    try:
+      func(logger=logger)
+      logger.info("success")
+    except BaseException as e:
+      success = False
+      logger.error(e)
+  if success:
+    result_file.write("HEALTHY\n")
```

Comment:
should we delete or truncate the file at start of run? i'm thinking of what happens if we write healthy at one point and then fail to write it again afterward.

Reply:
sgtm; updated to truncate the file.